### PR TITLE
connectd: check all chain hashs of the network array in `init`

### PR DESCRIPTION
**Pull request by darosior**
As per https://github.com/lightningnetwork/lightning-rfc/pull/682#issuecomment-566066644.

We would always check against the first element of the array.. Once again my bad.

This also cleans up a redundant check as [pointed out](https://github.com/ElementsProject/lightning/pull/3300#discussion_r352380389) by @rustyrussell.

Changelog-None

### CODE REVIEW

**Comment by rustyrussell** [comment]
Ack https://github.com/ElementsProject/lightning/commit/9fa93dfcddd05989fd324211e6e8aa89c7e78f7f

**Comment by rustyrussell** [comment]
Ack https://github.com/ElementsProject/lightning/commit/9fa93dfcddd05989fd324211e6e8aa89c7e78f7f

File: connectd/peer_exchange_initmsg.c [hunk 1]
```diff
@@ -75,12 +75,6 @@ static struct io_plan *peer_init_received(struct io_conn *conn,
 	 *    - MAY fail the connection.
 	 */
 	if (tlvs->networks) {
-		if (!tlvs->networks->chains) {
```
Comment (niftynei) [inline connectd/peer_exchange_initmsg.c]: nit: this change belongs in a separate commit from the first, as they're not related changes

**Comment by cdecker** [comment]
> Ack [9fa93df](https://github.com/ElementsProject/lightning/commit/9fa93dfcddd05989fd324211e6e8aa89c7e78f7f)

@rustyrussell you're offline, remember? :stuck_out_tongue_winking_eye:

**Comment by darosior** [comment]
Separated into two commits.. Looking at the test now

**Comment by darosior** [comment]
Could finally test it using protocol tests, will cleanup and submit a PR there ?

**Comment by darosior** [comment]
Finally I also added a commit to return an error instead of `io_close`ing the connection, which broke protocol tests.

Tests at https://github.com/ElementsProject/lightning-rfc-protocol-test/pull/13

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/3371/commits/d45c27935ad5b7085423e35ef1dd835029ec7f2c

**Comment by cdecker** [comment]
Only missing a test for this. I think the pyln-proto library might be candidate to inject the `init` message here.

**Comment by darosior** [comment]
@cdecker isn't the protocol test better to test this ? I at first wanted to use `pyln-proto`, but then prefered ElementsProject/lightning-rfc-protocol-test#13
